### PR TITLE
Trigger batch jobs using splice.

### PR DESCRIPTION
**Pull request by spxtr**
Sorry, this PR is pretty brutal.

Still need to test e2e, but I think this is the general idea.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/1168)

### CODE REVIEW

File: prow/cmd/deck/static/script.js [hunk 1]
```diff
@@ -144,8 +150,13 @@ function redraw() {
         var r = document.createElement("tr");
         r.appendChild(stateCell(allBuilds[i].state));
         r.appendChild(createLinkCell(allBuilds[i].repo, "https://github.com/" + allBuilds[i].repo));
-        r.appendChild(createLinkCell(allBuilds[i].number, "https://github.com/" + allBuilds[i].repo + "/pull/" + allBuilds[i].number));
-        r.appendChild(createLinkCell(allBuilds[i].author, "https://github.com/" + allBuilds[i].author));
+        if (allBuilds[i].type == "batch") {
+            r.appendChild(createTextCell("Batch"));
+            r.appendChild(createTextCell(""));
```
Comment (rmmh) [inline prow/cmd/deck/static/script.js]: needs a TODO

File: prow/cmd/splice/main.go [hunk 2]
```diff
@@ -169,27 +176,21 @@ func (s *splicer) gitRef(ref string) string {
...
-}
-
-type buildRef struct {
-	Repo string  `json:"repo"`
-	Base string  `json:"base_ref"`
-	PRs  []prRef `json:"prs"`
-}
-
 // Produce a struct describing the exact revisions that were merged together.
 // This is sent to Prow and used by the Submit Queue when deciding whether it
```
Comment (rmmh) [inline prow/cmd/splice/main.go]: reword this comment

File: prow/cmd/splice/main.go [hunk 3]
```diff
@@ -202,32 +203,64 @@ func main() {
...
-		buildRef := splicer.makeBuildRef(*repoName, batchPRs)
-		buf, err := json.Marshal(buildRef)
-		if err != nil {
-			log.WithError(err).Fatal("unable to marshal JSON")
+		buildReq := splicer.makeBuildRequest(*orgName, *repoName, batchPRs)
+		if len(batchPRs) < 2 {
+			continue
+		}
+		if len(batchPRs) > 5 {
+			batchPRs = batchPRs[:5]
```
Comment (rmmh) [inline prow/cmd/splice/main.go]: this doesn't need to be guarded by the conditonal

File: prow/cmd/splice/main.go [hunk 4]
```diff
@@ -202,32 +203,64 @@ func main() {
...
+		}
+		log.Infof("Batch PRs: %v", batchPRs)
+		for _, job := range ja.AllJobs(fmt.Sprintf("%s/%s", *orgName, *repoName)) {
+			if job.AlwaysRun {
+				if err := line.StartJob(kc, job.Name, buildReq); err != nil {
+					log.WithError(err).WithField("job", job.Name).Error("Error starting job.")
+				}
+			}
 		}
```
Comment (rmmh) [inline prow/cmd/splice/main.go]: wait for batch completion here?

File: prow/cmd/splice/main_test.go [hunk 5]
```diff
@@ -145,23 +144,3 @@ func TestFindMergeable(t *testing.T) {
 	mergeable, err = s.findMergeable(up.dir, []int{3, 2, 1, 4})
 	expectEqual(t, "mergeable PRs", mergeable, []int{3, 2, 1})
 }
-
-func TestMakeBuildRef(t *testing.T) {
```
Comment (rmmh) [inline prow/cmd/splice/main_test.go]: Do you think MakeBuildRequest is so trivial it doesn't need a test?

Comment (spxtr) [inline prow/cmd/splice/main_test.go]: Nearly so.

File: prow/cmd/splice/main.go (hunk 4 above)
Comment (spxtr) [inline prow/cmd/splice/main.go]: I wait at the top of the loop.

File: prow/cmd/splice/main.go [hunk 6]
```diff
@@ -202,32 +203,64 @@ func main() {
...
-		if err != nil {
-			log.WithError(err).Fatal("unable to marshal JSON")
+		buildReq := splicer.makeBuildRequest(*orgName, *repoName, batchPRs)
+		if len(batchPRs) < 2 {
+			continue
+		}
+		if len(batchPRs) > 5 {
+			batchPRs = batchPRs[:5]
+		}
+		log.Infof("Batch PRs: %v", batchPRs)
```
Comment (spxtr) [inline prow/cmd/splice/main.go]: Done.

File: prow/cmd/splice/main.go (hunk 3 above)
Comment (spxtr) [inline prow/cmd/splice/main.go]: Yes, go will panic if we pass that a length-4 slice.

File: prow/cmd/splice/main.go (hunk 2 above)
Comment (spxtr) [inline prow/cmd/splice/main.go]: Done.

File: prow/cmd/splice/main.go [hunk 7]
```diff
@@ -62,6 +68,7 @@ func getQueuedPRs(url string) ([]int, error) {
 	queue := struct {
 		E2EQueue []struct {
 			Number int
+			Login  string
```
Comment (spxtr) [inline prow/cmd/splice/main.go]: Done. At one point I was going to include author somewhere, but decided against it.

File: prow/cmd/deck/static/script.js (hunk 1 above)
Comment (spxtr) [inline prow/cmd/deck/static/script.js]: Done.

File: prow/cmd/deck/static/script.js [hunk 8]
```diff
@@ -130,6 +132,10 @@ function redraw() {
     var pr = document.getElementById("pr").value;
 
     for (var i = 0; i < allBuilds.length; i++) {
```
Comment (spxtr) [inline prow/cmd/deck/static/script.js]: Done.

**Comment by spxtr** [comment]
Will push tomorrow.

**Pull request merged**